### ISSUE TITLE
BUG: Can't execute example notebook

### ISSUE THREAD

**Issue by jannessm** [KitwareMedical/ITKUltrasound#125]
Hello,

I am very interested in using this package. But sadly, I was not able to execute the example notebook at all. First, I struggled with finding the correct python version to be able to install all required packages maybe it would be beneficial to add the required and tested python versions to your documentation.

And secondly, I noticed that your current version depends on a pre-release of itk. Why have you chosen to depend on an unstable release?

When executing the notebook an error occurs at cell 5 due to an import error in ITKCommonBasePython.py. Am I missing a package or is this due to an error in itk itself? I already create an [issue](https://github.com/InsightSoftwareConsortium/ITK/issues/2407) for itk, too.

```
in <module>
     25     from . import _ITKCommonBasePython
     26 else:
---> 27     import _ITKCommonBasePython
     28 
     29 try:

ModuleNotFoundError: No module named '_ITKCommonBasePython'
```

I am currently working on a MacOS system with anaconda. 

Thank you!

**Comment by jannessm** [comment]
Here are all my versions from the current env:

```
itk                             5.2rc3
itk-bsplinegradient             0.2.3
itk-core                        5.2rc3
itk-filtering                   5.2rc3
itk-higherorderaccurategradient 1.0.8
itk-io                          5.2rc3
itk-numerics                    5.2rc3
itk-registration                5.2rc3
itk-segmentation                5.2rc3
itk-splitcomponents             2.0.1
itk-strain                      0.3.1
itk-ultrasound                  0.3
```

**Comment by dzenanz** [comment]
We produced the latest packages last week, so they have definitely been under-tested. Due to a mistake on my part, the current packages are not available with Python 3.7 on MacOS (only 3.6, 3.8 and 3.9). 3.7 should be included in the next iteration. ITK 5.2 final is scheduled for release this month, and RC3 should be very close to the final one, so we are in part testing ITK itself too.

I don't think you are missing a package. Our Python wrapping might be missing for a few classes. We are working on this now, please stay tuned while we investigate/fix this.

**Comment by jannessm** [comment]
Great. Thank you for the response!

But while testing I noticed that for 3.8 and 3.9 aren't available, too. The problem is that no version for itk-splitcomponents can be found. Maybe it is because version 2.0.2 wasn't published yet.

Regards.

**Comment by thewtex** [comment]
@jannessm yes, this was related to dependency versions.  Could you please try with `itk-ultrasound=0.3.1` in a clean virtualenv or conda environment?

**Comment by jannessm** [comment]
I have tested now version 0.3.1 in a conda env. with the following versions:

```
itk                       5.2rc3                   pypi_0    pypi
itk-bsplinegradient       0.2.3                    pypi_0    pypi
itk-core                  5.2rc3                   pypi_0    pypi
itk-filtering             5.2rc3                   pypi_0    pypi
itk-higherorderaccurategradient 1.0.8                    pypi_0    pypi
itk-io                    5.2rc3                   pypi_0    pypi
itk-numerics              5.2rc3                   pypi_0    pypi
itk-registration          5.2rc3                   pypi_0    pypi
itk-segmentation          5.2rc3                   pypi_0    pypi
itk-splitcomponents       2.0.3                    pypi_0    pypi
itk-strain                0.3.1                    pypi_0    pypi
itk-ultrasound            0.3.1                    pypi_0    pypi
```

Sadly, I get now different errors and warnings.

For cell 5:

```
Warning: Unknown parameter 'itk::CurvilinearArraySpecialCoordinatesImage<std::complex<double>,2>' in template 'itk::BModeImageFilter'
```

for cell 10:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-9-6e2ef854dd2a> in <module>
      2 gain = itk.Array2D[itk.D](num_gain_controls, 2)
      3 for ii in np.arange(num_gain_controls):
----> 4     gain.SetElement(ii, 0, float(ii) / (num_gain_controls - 1) * aline_samples)
      5     gain.SetElement(ii, 1, 1.0)
      6 tgc_filter.SetGain(gain)

TypeError: in method 'itkArray2DD_SetElement', argument 2 of type 'unsigned long'
```

For cell 15, 19, and 20:

```
RuntimeError: ../../../include/itkTimeGainCompensationImageFilter.hxx:73:
itk::ERROR: itk::ERROR: TimeGainCompensationImageFilter(0x7faac83984c0): Gain depths must be strictly increasing.
```

**Comment by thewtex** [comment]
@jannessm thanks for the report :book: :eyes: 

> Warning: Unknown parameter 'itk::CurvilinearArraySpecialCoordinatesImage<std::complex<double>,2>' in template 'itk::BModeImageFilter'

We will address this re: #126 

> ----> 4     gain.SetElement(ii, 0, float(ii) / (num_gain_controls - 1) * aline_samples)

Please try pulling the latest version of the notebook. It should be:

```
for ii in np.arange(num_gain_controls):
    gain.SetElement(int(ii), 0, float(ii) / (num_gain_controls - 1) * aline_samples)
    gain.SetElement(int(ii), 1, 1.0)
```

**Comment by jannessm** [comment]
@thewtex Thank you for helping me execute your example. Sadly, after updating the notebook and installing all latest packages, I get an import error in cell 4.

```python
ImportError                               Traceback (most recent call last)
<ipython-input-4-06d643b6498c> in <module>
      2 Dimension = 2
      3 IntegerPixelType = itk.SS
----> 4 IntegerImageType = itk.CurvilinearArraySpecialCoordinatesImage[IntegerPixelType, Dimension]
      5 RealPixelType = itk.F
      6 RealImageType = itk.CurvilinearArraySpecialCoordinatesImage[RealPixelType, Dimension]

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/support/lazy.py in __getattribute__(self, attr)
     74                     module = self.__belong_lazy_attributes[attr]
     75                     namespace = {}
---> 76                     base.itk_load_swig_module(module, namespace)
     77                     self.loaded_lazy_modules.add(module)
     78                     for k, v in namespace.items():

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/support/base.py in itk_load_swig_module(name, namespace)
    100         deps = l_data.get_module_dependencies()
    101         for dep in deps:
--> 102             itk_load_swig_module(dep, namespace)
    103 
    104     if itkConfig.ImportCallback:

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/support/base.py in itk_load_swig_module(name, namespace)
    108     # if we haven't already.
    109     loader = LibraryLoader()
--> 110     l_module = loader.load(swig_module_name)
    111 
    112     # OK, now the modules on which this one depends are loaded and

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/support/base.py in load(self, name)
    257             # since version 3.4: Use importlib.util.find_spec() instead.
    258             l_spec = importlib.util.find_spec(name)
--> 259             l_spec.loader.exec_module(l_module)  # pytype: disable=attribute-error
    260             return l_module
    261         finally:

~/miniconda3/envs/itk/lib/python3.9/importlib/_bootstrap_external.py in exec_module(self, module)

~/miniconda3/envs/itk/lib/python3.9/importlib/_bootstrap.py in _call_with_frames_removed(f, *args, **kwds)

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/support/../StrainPython.py in <module>
     71 import itk.ITKImageGradientPython
     72 import itk.ITKCommonPython
---> 73 from itk.itkStrainImageFilterPython import *
     74 
     75 

~/miniconda3/envs/itk/lib/python3.9/site-packages/itk/itkStrainImageFilterPython.py in <module>
    441 
    442 
--> 443 from itk.support import itkHelpers
    444 @itkHelpers.accept_numpy_array_like_xarray
    445 def strain_image_filter(*args, **kwargs):

ImportError: cannot import name 'itkHelpers' from 'itk.support' (unknown location)
```

with the following itk packages:

```
itk                             5.2.0
itk-bsplinegradient             0.2.3
itk-core                        5.2.0
itk-filtering                   5.2.0
itk-higherorderaccurategradient 1.0.8
itk-io                          5.2.0
itk-numerics                    5.2.0
itk-registration                5.2.0
itk-segmentation                5.2.0
itk-splitcomponents             2.0.3
itk-strain                      0.3.1
itk-ultrasound                  0.3.1
```

**Comment by dzenanz** [comment]
This issue now also appears in PR #131, unexpectedly.

**Comment by thewtex** [comment]
Addressed with `itk-ultrasound-0.3.2` as verified by the CI in https://github.com/KitwareMedical/ITKUltrasound/pull/134

**Comment by thewtex** [comment]
We now also have `itk-ultrasound-0.4.0`.

@jannessm if the issue persists, please re-open, or if please let use know other problems by opening additional issues as they are encountered. Thanks!
